asciidoctorj-core/src/test/groovy/org/asciidoctor/converter/WhenFootnotesAreUsed.groovy

```diff
@@ -0,0 +1,116 @@
+package org.asciidoctor.converter
+
+import org.asciidoctor.Asciidoctor
+import org.asciidoctor.OptionsBuilder
+import org.asciidoctor.ast.ContentNode
+import org.asciidoctor.ast.Document
+import org.asciidoctor.ast.Footnote
+import org.asciidoctor.ast.StructuralNode
+import org.asciidoctor.jruby.ast.impl.FootnoteImpl
+import org.jboss.arquillian.spock.ArquillianSputnik
+import org.jboss.arquillian.test.api.ArquillianResource
+import org.junit.runner.RunWith
+import spock.lang.Specification
+
+import static org.hamcrest.Matchers.contains
+import static org.hamcrest.Matchers.empty
+import static org.hamcrest.Matchers.samePropertyValuesAs
+import static org.junit.Assert.assertThat
+
+/**
+ * Tests that footnotes can be accessed from converter.
+ *
+ * Note that it is a current limitation of asciidoctor that footnotes are not available until
+ * after they have been converted for the document.
+ */
+@RunWith(ArquillianSputnik)
+class WhenFootnotesAreUsed extends Specification {
+    static final String CONVERTER_BACKEND = 'footnote'
+
+    @ArquillianResource
+    private Asciidoctor asciidoctor
+    private static List<Footnote> footnotesBeforeConvert
```

Comment:
To make test-methods fully indepent from eachother we could convert these into class attributes of the Converter and register a different instance with something like:

```java
asciidoctor.javaConverterRegistry().register(new Converter(...), CONVERTER_BACKEND)
```

Reply:
I don't see a `register` that takes an instance, only one that takes a class.

I suppose I could `unregisterAll` as part of a test tear down.

Reply:
Omg is true! I assumes because extensions have this options.
Then just ignore the comment, no need to overcomplicate things and unregistering won't fix concurrency issues.

We may need to address this in another PR.

Reply:
No problemo, thanks for the continued comments and reviews, I really appreciate them.